### PR TITLE
Merge release-1.1 to master

### LINKED ISSUE #2218: master: sync release 1.1 branch
- [ ] Update istio/api for 1.1rc5 (#2153)
a5d5a46

- [ ] Fixes environment-dependent failures in MixerFaultTest (#2144)
9d52640

- [ ] Update API and import envoyproxy/envoy#6263 (#2157)  …
55c8096

- [ ] update envoy to latest (#2158)  …
5a9945b
 
- [ ] release 1.1: Update envoy to enable path normalization (#2166)  …
bf8b9cd

- [ ] Sync istio/api for 1.1.3 (#2170)
a169a0c

- [ ] Updating API sha for proxy. (#2181)  …
73fa9b1

- [ ] Retrieve tracing header from request headers during report (#2182)  …
56242cd

### PR DESCRIPTION
**Pull request by crazyxy**
**What this PR does / why we need it**: Merger release-1.1 to master

**Which issue this PR fixes**: fixes #2218

### CODE REVIEW

**Comment by silentdai** [comment]
@crazyxy  You are almost there! 
The build is failing because the 1.1 branch is using old header map API.
To fix it you can see the example here  https://github.com/istio/proxy/pull/2179

**Comment by crazyxy** [comment]
@duderino need your help to merge the pr

**Comment by duderino** [comment]
I will work with @crazyxy on Monday to do the merge on the command line.  If I merge it via the github UI I will squash all the commits and lose the git history.

File: src/envoy/utils/utils.cc [hunk 1]
```diff
@@ -36,21 +37,21 @@ const std::string kMetadataDestinationUID("uid");
 
 }  // namespace
 
-void ExtractHeaders(const Http::HeaderMap& header_map,
-                    const std::set<std::string>& exclusives,
-                    std::map<std::string, std::string>& headers) {
+void ExtractHeaders(const Http::HeaderMap &header_map,
```
Comment (silentdai) [inline src/envoy/utils/utils.cc]: Could you check when the "&" "*" format is introduced? I don't see this pattern else where

File: WORKSPACE [hunk 2]
```diff
@@ -32,6 +32,9 @@ bind(
     actual = "//external:ssl",
 )
 
+# envoy commit date 04/10/2019
```
Comment (silentdai) [inline WORKSPACE]: I doubt some of the conflicts are not resolved correctly. John's PR shoudl beat this one: https://github.com/istio/proxy/commit/0b76a30145d2a67429758ebe7c34672abd9e0498

File: src/envoy/utils/utils.cc (hunk 1 above)
Comment (crazyxy) [reply src/envoy/utils/utils.cc]: I doubt this is caused by the different version of `clang-format` I used.

File: WORKSPACE (hunk 2 above)
Comment (crazyxy) [reply WORKSPACE]: This is the redundant message. John's PR is down. If you check line 41, you will find John's PR. But I don't know why I didn't find this line when I resolving conflicts.

File: src/envoy/utils/utils.cc (hunk 1 above)
Comment (silentdai) [reply src/envoy/utils/utils.cc]: If it is introduced at your merge, you'd better fixed with this PR.
If it is introduced by existing PR, we could fix it first and merge to master

File: WORKSPACE (hunk 2 above)
Comment (silentdai) [reply WORKSPACE]: Sure. git is not that reliable :) That's why we have this review 
Please fix it in the follow up

**Pull request closed without merging**